referenceframe/frame_json.go

```diff
@@ -199,16 +199,19 @@ func jsonToFrame(data json.RawMessage) (Frame, error) {
 	if err := json.Unmarshal(sF["frame_type"], &frameType); err != nil {
 		return nil, err
 	}
+	if _, ok := sF["frame"]; !ok {
+		return nil, fmt.Errorf("no frame data found for frame, type was %s", frameType)
+	}
 
 	implementer, ok := registeredFrameImplementers[frameType]
 	if !ok {
 		return nil, fmt.Errorf("%s is not a registered Frame implementation", frameType)
 	}
 	frameZeroStruct := reflect.New(implementer).Elem()
-	if err := json.Unmarshal(sF["frame"], frameZeroStruct.Addr().Interface()); err != nil {
+	frame := frameZeroStruct.Addr().Interface()
```

Comment:
1. Are these two lines the same thing as:
```
frame := reflect.Zero(implementer).Interface()
```

If so, could we go with that implementation as it makes the intent a lot clearer? 

2. Could we please do the `utils.AssertType[Frame](frame)` before we call `json.Unmarshnal` as otherwise we don't know what code the unmarshaler is going to call if the type is not a Frame.

Reply:
Handled #2, but sadly it appears that #1 is not equivalent